airflow/providers/databricks/hooks/databricks.py

```diff
@@ -73,7 +72,9 @@ def is_successful(self) -> bool:
         """True if the result state is SUCCESS"""
         return self.result_state == 'SUCCESS'
 
-    def __eq__(self, other) -> bool:
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, RunState):
+            return NotImplemented
```

Comment:
Should it be raise?

Reply:
It's a special value, which should be returned by binary special methods, we could have a raise for `NotImplementedError`. 

More info here: https://docs.python.org/3/library/constants.html#NotImplemented